core/node/fee_model/src/l1_gas_price/blob_base_fee_predictor.rs

```diff
@@ -0,0 +1,113 @@
+use zksync_dal::{Connection, Core, CoreDal};
+use zksync_types::{
+    aggregated_operations::{AggregatedActionType, L1BatchAggregatedActionType},
+    web3::{BlockId, BlockNumber},
+    L1BatchNumber,
+};
+
+use crate::l1_gas_price::GasAdjusterClient;
+
+pub(crate) async fn predict_blob_base_fee(
+    connection: &mut Connection<'_, Core>,
+    client: &GasAdjusterClient,
+    last_known_l1_gas_price: u64,
+) -> u64 {
+    let batch_stats = connection
+        .eth_sender_dal()
+        .get_eth_all_blocks_stat()
+        .await
+        .expect("Failed to get batch stats");
+    let mut last_l1_commited_batch: Option<L1BatchNumber> = None;
+
+    for (tx_type, block) in batch_stats.mined {
+        if tx_type == AggregatedActionType::L1Batch(L1BatchAggregatedActionType::Commit) {
+            last_l1_commited_batch = Some(block.into());
+        }
+    }
+
+    let last_l1_commited_batch = last_l1_commited_batch.unwrap_or(L1BatchNumber::from(0));
+    let last_sealed_batch = connection
+        .blocks_dal()
+        .get_sealed_l1_batch_number()
+        .await
+        .expect("Failed to get last sealed batch")
+        .unwrap_or(L1BatchNumber::from(0));
+
+    let total_blobs_to_send = connection
+        .blocks_dal()
+        .get_blobs_amount_for_range(last_l1_commited_batch + 1, last_sealed_batch)
+        .await
+        .expect("Failed to get blobs amount for range");
+
+    let latest_block_number = client
+        .inner
+        .block(BlockId::Number(BlockNumber::Latest))
+        .await
+        .expect("Failed to get latest block")
+        .expect("Latest block is None")
+        .number
+        .expect("Latest block number is None");
+    let last_commited_block_number = connection
+        .eth_sender_dal()
+        .get_sent_at_block_for_commited_block(last_l1_commited_batch)
+        .await
+        .expect("Failed to get sent at block for commited block")
+        .unwrap_or(latest_block_number.as_u32());
+
+    let mut total_l1_blocks_for_these_blocks = latest_block_number
+        .saturating_sub(last_commited_block_number.into())
```

Comment:
> last_commited_block_number

may belong to either GW or L1, so please be careful here, the formula should be only used for L1